diff --git a/configure.ac b/configure.ac
index cc592e1..a3cbff2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -59,6 +59,14 @@ AM_CONDITIONAL([HAVE_PTHREADS],[test "x$have_pthreads" = "xyes"])
 AC_SUBST(PTHREAD_CFLAGS)
 AC_SUBST(PTHREAD_LIBS)
 
+# OpenCPI wrappers:
+OS_VERSION
+AC_CANONICAL_HOST
+AS_CASE([$host_os],[*darwin*],[HOSTOS="macos"],[*linux*],[HOSTOS="linux"])
+AC_SUBST([HOSTOS])
+AS_CASE([$host_cpu],[x86_64],[HOSTARCH="x86_64"],[i?86],[HOSTARCH="x86"],[arm*],[HOSTARCH="arm"])
+AC_SUBST([HOSTARCH])
+m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 # TODO(chandlerc@google.com) Check for the necessary system headers.
 
 # TODO(chandlerc@google.com) Check the types, structures, and other compiler
diff -rupN gtest-1.7.0.org/m4/osver.m4 gtest-1.7.0/m4/osver.m4
--- gtest-1.7.0.org/m4/osver.m4	1969-12-31 19:00:00.000000000 -0500
+++ gtest-1.7.0/m4/osver.m4	2015-04-13 14:26:48.778352476 -0400
@@ -0,0 +1,49 @@
+AC_DEFUN([OS_VERSION],
+[
+	AC_CACHE_CHECK([os version], os_cv_os_ver, [
+		AC_ARG_WITH(os-ver,
+		AC_HELP_STRING([--with-os-ver], [host os version (default from environment variable 'TARGET_OS_VERSION')]),
+		os_cv_os_ver=$withval,
+        if test "x${TARGET_OS_VERSION}" != "x"; then
+			os_cv_os_ver=${TARGET_OS_VERSION}
+		else 
+		if test -f /etc/redhat-release; then
+        # redhat/centos is just based on the major release number,
+        # with minor releases assumed to be binary compatible.
+        temp=$(cat /etc/redhat-release)       
+        temp2=$( sed -e 's/\([0-9]\)\..*/\1/' /etc/redhat-release)
+        v2=$(sed -e 's/^@<:@^0-9@:>@*\(@<:@0-9@:>@\+\)\..*$/\1/' /etc/redhat-release)
+        v3=$(echo $v2 | sed 's/\ (Final)//')
+        v1=$(echo "$temp" | sed 's/^\(.\).*/\1/' |  tr A-Z a-z)         
+        if test "$temp" = ""; then
+        echo Cannot parse redhat/centos release from /etc/redhat-release 1>&2
+    exit 1
+  fi
+  os_cv_os_ver=$v1$v3
+elif test -f /etc/lsb-release; then
+  . /etc/lsb-release
+  os_cv_os_ver=`echo $DISTRIB_ID|sed 's/^\(.\).*/\1/' | tr A-Z a-z``echo $DISTRIB_RELEASE|sed 's/^\([^.]*\).*/\1/'`
+elif test -f /etc/os-release; then
+  # We'll assume debian
+  . /etc/os-release
+  os_cv_os_ver=d$VERSION_ID
+elif test $HostSystem = darwin -a "`which sw_vers`" != ""; then
+  os_cv_os_ver=`sw_vers -productVersion | sed 's/^\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/' | tr . _`
+elif test "$HostSystem" = linux -a -f /etc/rootfs_version; then
+  os_cv_os_ver=iv
+elif test "$HostSystem" = linux -a "$HostProcessor" = armv7l; then
+  os_cv_os_ver=zynq
+else
+  echo Cannot determine runtime host'!!' 1>&2
+  exit 1
+fi
+		fi
+		)
+	dnl check if this is a cross compile session, prepend the sysroot
+	AS_IF([test "x$cross_compiling" = "xyes"],[
+	CROSS_SYSROOT=`$CC --print-sysroot`
+	os_os_ver=${CROSS_SYSROOT}${os_cv_os_ver}
+	])
+	])
+AC_SUBST(os_ver, $os_cv_os_ver)
+])
diff --git a/Makefile.am b/Makefile.am
index 29797e4..a67f7b2 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -2,170 +2,9 @@
 
 ACLOCAL_AMFLAGS = -I m4
 
-# Nonstandard package files for distribution
-EXTRA_DIST = \
-  CHANGES \
-  CONTRIBUTORS \
-  LICENSE \
-  include/gtest/gtest-param-test.h.pump \
-  include/gtest/internal/gtest-param-util-generated.h.pump \
-  include/gtest/internal/gtest-tuple.h.pump \
-  include/gtest/internal/gtest-type-util.h.pump \
-  make/Makefile \
-  scripts/fuse_gtest_files.py \
-  scripts/gen_gtest_pred_impl.py \
-  scripts/pump.py \
-  scripts/test/Makefile
-
-# gtest source files that we don't compile directly.  They are
-# #included by gtest-all.cc.
-GTEST_SRC = \
-  src/gtest-death-test.cc \
-  src/gtest-filepath.cc \
-  src/gtest-internal-inl.h \
-  src/gtest-port.cc \
-  src/gtest-printers.cc \
-  src/gtest-test-part.cc \
-  src/gtest-typed-test.cc \
-  src/gtest.cc
-
-EXTRA_DIST += $(GTEST_SRC)
-
-# Sample files that we don't compile.
-EXTRA_DIST += \
-  samples/prime_tables.h \
-  samples/sample2_unittest.cc \
-  samples/sample3_unittest.cc \
-  samples/sample4_unittest.cc \
-  samples/sample5_unittest.cc \
-  samples/sample6_unittest.cc \
-  samples/sample7_unittest.cc \
-  samples/sample8_unittest.cc \
-  samples/sample9_unittest.cc
-
-# C++ test files that we don't compile directly.
-EXTRA_DIST += \
-  test/gtest-death-test_ex_test.cc \
-  test/gtest-death-test_test.cc \
-  test/gtest-filepath_test.cc \
-  test/gtest-linked_ptr_test.cc \
-  test/gtest-listener_test.cc \
-  test/gtest-message_test.cc \
-  test/gtest-options_test.cc \
-  test/gtest-param-test2_test.cc \
-  test/gtest-param-test2_test.cc \
-  test/gtest-param-test_test.cc \
-  test/gtest-param-test_test.cc \
-  test/gtest-param-test_test.h \
-  test/gtest-port_test.cc \
-  test/gtest_premature_exit_test.cc \
-  test/gtest-printers_test.cc \
-  test/gtest-test-part_test.cc \
-  test/gtest-tuple_test.cc \
-  test/gtest-typed-test2_test.cc \
-  test/gtest-typed-test_test.cc \
-  test/gtest-typed-test_test.h \
-  test/gtest-unittest-api_test.cc \
-  test/gtest_break_on_failure_unittest_.cc \
-  test/gtest_catch_exceptions_test_.cc \
-  test/gtest_color_test_.cc \
-  test/gtest_env_var_test_.cc \
-  test/gtest_environment_test.cc \
-  test/gtest_filter_unittest_.cc \
-  test/gtest_help_test_.cc \
-  test/gtest_list_tests_unittest_.cc \
-  test/gtest_main_unittest.cc \
-  test/gtest_no_test_unittest.cc \
-  test/gtest_output_test_.cc \
-  test/gtest_pred_impl_unittest.cc \
-  test/gtest_prod_test.cc \
-  test/gtest_repeat_test.cc \
-  test/gtest_shuffle_test_.cc \
-  test/gtest_sole_header_test.cc \
-  test/gtest_stress_test.cc \
-  test/gtest_throw_on_failure_ex_test.cc \
-  test/gtest_throw_on_failure_test_.cc \
-  test/gtest_uninitialized_test_.cc \
-  test/gtest_unittest.cc \
-  test/gtest_unittest.cc \
-  test/gtest_xml_outfile1_test_.cc \
-  test/gtest_xml_outfile2_test_.cc \
-  test/gtest_xml_output_unittest_.cc \
-  test/production.cc \
-  test/production.h
-
-# Python tests that we don't run.
-EXTRA_DIST += \
-  test/gtest_break_on_failure_unittest.py \
-  test/gtest_catch_exceptions_test.py \
-  test/gtest_color_test.py \
-  test/gtest_env_var_test.py \
-  test/gtest_filter_unittest.py \
-  test/gtest_help_test.py \
-  test/gtest_list_tests_unittest.py \
-  test/gtest_output_test.py \
-  test/gtest_output_test_golden_lin.txt \
-  test/gtest_shuffle_test.py \
-  test/gtest_test_utils.py \
-  test/gtest_throw_on_failure_test.py \
-  test/gtest_uninitialized_test.py \
-  test/gtest_xml_outfiles_test.py \
-  test/gtest_xml_output_unittest.py \
-  test/gtest_xml_test_utils.py
-
-# CMake script
-EXTRA_DIST += \
-  CMakeLists.txt \
-  cmake/internal_utils.cmake
-
-# MSVC project files
-EXTRA_DIST += \
-  msvc/gtest-md.sln \
-  msvc/gtest-md.vcproj \
-  msvc/gtest.sln \
-  msvc/gtest.vcproj \
-  msvc/gtest_main-md.vcproj \
-  msvc/gtest_main.vcproj \
-  msvc/gtest_prod_test-md.vcproj \
-  msvc/gtest_prod_test.vcproj \
-  msvc/gtest_unittest-md.vcproj \
-  msvc/gtest_unittest.vcproj
-
-# xcode project files
-EXTRA_DIST += \
-  xcode/Config/DebugProject.xcconfig \
-  xcode/Config/FrameworkTarget.xcconfig \
-  xcode/Config/General.xcconfig \
-  xcode/Config/ReleaseProject.xcconfig \
-  xcode/Config/StaticLibraryTarget.xcconfig \
-  xcode/Config/TestTarget.xcconfig \
-  xcode/Resources/Info.plist \
-  xcode/Scripts/runtests.sh \
-  xcode/Scripts/versiongenerate.py \
-  xcode/gtest.xcodeproj/project.pbxproj
-
-# xcode sample files
-EXTRA_DIST += \
-  xcode/Samples/FrameworkSample/Info.plist \
-  xcode/Samples/FrameworkSample/WidgetFramework.xcodeproj/project.pbxproj \
-  xcode/Samples/FrameworkSample/runtests.sh \
-  xcode/Samples/FrameworkSample/widget.cc \
-  xcode/Samples/FrameworkSample/widget.h \
-  xcode/Samples/FrameworkSample/widget_test.cc
-
-# C++Builder project files
-EXTRA_DIST += \
-  codegear/gtest.cbproj \
-  codegear/gtest.groupproj \
-  codegear/gtest_all.cc \
-  codegear/gtest_link.cc \
-  codegear/gtest_main.cbproj \
-  codegear/gtest_unittest.cbproj
-
 # Distribute and install M4 macro
 m4datadir = $(datadir)/aclocal
 m4data_DATA = m4/gtest.m4
-EXTRA_DIST += $(m4data_DATA)
 
 # We define the global AM_CPPFLAGS as everything we compile includes from these
 # directories.
@@ -196,7 +35,6 @@ pkginclude_HEADERS = \
   include/gtest/gtest_pred_impl.h \
   include/gtest/gtest_prod.h
 
-pkginclude_internaldir = $(pkgincludedir)/internal
 pkginclude_internal_HEADERS = \
   include/gtest/internal/gtest-death-test-internal.h \
   include/gtest/internal/gtest-filepath.h \
@@ -208,7 +46,9 @@ pkginclude_internal_HEADERS = \
   include/gtest/internal/gtest-port-arch.h \
   include/gtest/internal/gtest-string.h \
   include/gtest/internal/gtest-tuple.h \
-  include/gtest/internal/gtest-type-util.h \
+  include/gtest/internal/gtest-type-util.h
+
+pkginclude_internal_custom_HEADERS = \
   include/gtest/internal/custom/gtest.h \
   include/gtest/internal/custom/gtest-port.h \
   include/gtest/internal/custom/gtest-printers.h
@@ -216,95 +56,9 @@ pkginclude_internal_HEADERS = \
 lib_libgtest_main_la_SOURCES = src/gtest_main.cc
 lib_libgtest_main_la_LIBADD = lib/libgtest.la
 
-# Bulid rules for samples and tests. Automake's naming for some of
-# these variables isn't terribly obvious, so this is a brief
-# reference:
-#
-# TESTS -- Programs run automatically by "make check"
-# check_PROGRAMS -- Programs built by "make check" but not necessarily run
-
-noinst_LTLIBRARIES = samples/libsamples.la
-
-samples_libsamples_la_SOURCES = \
-  samples/sample1.cc \
-  samples/sample1.h \
-  samples/sample2.cc \
-  samples/sample2.h \
-  samples/sample3-inl.h \
-  samples/sample4.cc \
-  samples/sample4.h
-
-TESTS=
-TESTS_ENVIRONMENT = GTEST_SOURCE_DIR="$(srcdir)/test" \
-                    GTEST_BUILD_DIR="$(top_builddir)/test"
-check_PROGRAMS=
-
-# A simple sample on using gtest.
-TESTS += samples/sample1_unittest
-check_PROGRAMS += samples/sample1_unittest
-samples_sample1_unittest_SOURCES = samples/sample1_unittest.cc
-samples_sample1_unittest_LDADD = lib/libgtest_main.la \
-                                 lib/libgtest.la \
-                                 samples/libsamples.la
-
-# Another sample.  It also verifies that libgtest works.
-TESTS += samples/sample10_unittest
-check_PROGRAMS += samples/sample10_unittest
-samples_sample10_unittest_SOURCES = samples/sample10_unittest.cc
-samples_sample10_unittest_LDADD = lib/libgtest.la
-
-# This tests most constructs of gtest and verifies that libgtest_main
-# and libgtest work.
-TESTS += test/gtest_all_test
-check_PROGRAMS += test/gtest_all_test
-test_gtest_all_test_SOURCES = test/gtest_all_test.cc
-test_gtest_all_test_LDADD = lib/libgtest_main.la \
-                            lib/libgtest.la
-
-# Tests that fused gtest files compile and work.
-FUSED_GTEST_SRC = \
-  fused-src/gtest/gtest-all.cc \
-  fused-src/gtest/gtest.h \
-  fused-src/gtest/gtest_main.cc
-
-if HAVE_PYTHON
-TESTS += test/fused_gtest_test
-check_PROGRAMS += test/fused_gtest_test
-test_fused_gtest_test_SOURCES = $(FUSED_GTEST_SRC) \
-                                samples/sample1.cc samples/sample1_unittest.cc
-test_fused_gtest_test_CPPFLAGS = -I"$(srcdir)/fused-src"
-
-# Build rules for putting fused Google Test files into the distribution
-# package. The user can also create those files by manually running
-# scripts/fuse_gtest_files.py.
-$(test_fused_gtest_test_SOURCES): fused-gtest
-
-fused-gtest: $(pkginclude_HEADERS) $(pkginclude_internal_HEADERS) \
-             $(GTEST_SRC) src/gtest-all.cc src/gtest_main.cc \
-             scripts/fuse_gtest_files.py
-	mkdir -p "$(srcdir)/fused-src"
-	chmod -R u+w "$(srcdir)/fused-src"
-	rm -f "$(srcdir)/fused-src/gtest/gtest-all.cc"
-	rm -f "$(srcdir)/fused-src/gtest/gtest.h"
-	"$(srcdir)/scripts/fuse_gtest_files.py" "$(srcdir)/fused-src"
-	cp -f "$(srcdir)/src/gtest_main.cc" "$(srcdir)/fused-src/gtest/"
-
-maintainer-clean-local:
-	rm -rf "$(srcdir)/fused-src"
-endif
-
-# Death tests may produce core dumps in the build directory. In case
-# this happens, clean them to keep distcleancheck happy.
-CLEANFILES = core
-
-# Disables 'make install' as installing a compiled version of Google
-# Test can lead to undefined behavior due to violation of the
-# One-Definition Rule.
-
-install-exec-local:
-	echo "'make install' is dangerous and not supported. Instead, see README for how to integrate Google Test into your build system."
-	false
-
-install-data-local:
-	echo "'make install' is dangerous and not supported. Instead, see README for how to integrate Google Test into your build system."
-	false
+# OpenCPI does make install with these:
+libdir=$(prefix)/$(HOSTOS)-$(os_ver)-$(HOSTARCH)/lib
+incdir=$(prefix)/include/gtest
+pkgincludedir=$(incdir)
+pkginclude_internaldir=$(incdir)/internal
+pkginclude_internal_customdir=$(incdir)/internal/custom
