diff -rupN xz-5.0.5_orig/configure.ac xz-5.0.5/configure.ac
--- xz-5.0.5_orig/configure.ac	2015-08-24 07:28:04.328774831 -0400
+++ xz-5.0.5/configure.ac	2015-08-24 08:13:26.709328876 -0400
@@ -15,7 +15,7 @@
 # of malloc(), stat(), or lstat(), since we don't use those functions in
 # a way that would cause the problems the autoconf macros check.
 
-AC_PREREQ([2.64])
+AC_PREREQ([2.63])
 
 AC_INIT([XZ Utils], m4_esyscmd([/bin/sh build-aux/version.sh]),
 	[lasse.collin@tukaani.org], [xz], [http://tukaani.org/xz/])
@@ -24,11 +24,24 @@ AC_CONFIG_AUX_DIR([build-aux])
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_HEADER([config.h])

+dnl FIXME remove when we assume autoconf 2.64
+m4_ifndef([AC_PACKAGE_URL],
+         [AC_DEFINE([PACKAGE_URL], ["http://tukaani.org/xz/"],
+                    [Define to home page for this package])
+          AC_SUBST([PACKAGE_URL], ["http://tukaani.org/xz/"])])
+
+
 echo
 echo "$PACKAGE_STRING"
 
 echo
 echo "System type:"
+OS_VERSION
+AS_CASE([$host_os],[*darwin*],[HOSTOS="macos"],[*linux*],[HOSTOS="linux"])
+AC_SUBST([HOSTOS])
+AS_CASE([$host_cpu],[x86_64],[HOSTARCH="x86_64"],[i?86],[HOSTARCH="x86"],[arm*],[HOSTARCH="arm"])
+AC_SUBST([HOSTARCH])
+
 # This is needed to know if assembler optimizations can be used.
 AC_CANONICAL_HOST
 
@@ -534,7 +547,8 @@
 # http://debbugs.gnu.org/cgi/bugreport.cgi?bug=17354
 # The -Wno-unsupported is used to silence warnings about missing
 # "subdir-objects".
-AM_INIT_AUTOMAKE([1.12 foreign tar-v7 filename-length-max=99 serial-tests -Wno-unsupported])
+AM_INIT_AUTOMAKE([1.11 foreign tar-v7 filename-length-max=99])
+m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 AC_PROG_LN_S
 
 AC_PROG_CC_C99
@@ -609,7 +623,7 @@

 echo
 echo "Initializing gettext:"
-AM_GNU_GETTEXT_VERSION([0.19])
+AM_GNU_GETTEXT_VERSION([0.17])
 AM_GNU_GETTEXT([external])


diff -rupN xz-5.0.5_orig/m4/osver.m4 xz-5.0.5/m4/osver.m4
--- xz-5.0.5_orig/m4/osver.m4	1969-12-31 19:00:00.000000000 -0500
+++ xz-5.0.5/m4/osver.m4	2015-08-24 07:35:27.730465250 -0400
@@ -0,0 +1,51 @@
+AC_DEFUN([OS_VERSION],
+[
+	AC_CACHE_CHECK([os version], os_cv_os_ver, [
+		AC_ARG_WITH(os-ver,
+		AC_HELP_STRING([--with-os-ver], [host os version (default from environment variable 'TARGET_OS_VERSION', options: c6/c7/zynq/rhel default is'c6')]),
+		os_cv_os_ver=$withval,
+        if test "x${TARGET_OS_VERSION}" != "x"; then
+			os_cv_os_ver=${TARGET_OS_VERSION}
+		else 
+		if test -f /etc/redhat-release; then
+        # redhat/centos is just based on the major release number,
+        # with minor releases assumed to be binary compatible.
+        temp=$(cat /etc/redhat-release)       
+        temp2=$( sed -e 's/\([0-9]\)\..*/\1/' /etc/redhat-release)
+        v2=$(sed -e 's/^@<:@^0-9@:>@*\(@<:@0-9@:>@\+\)\..*$/\1/' /etc/redhat-release)
+        v3=$(echo $v2 | sed 's/\ (Final)//')
+        v1=$(echo "$temp" | sed 's/^\(.\).*/\1/' |  tr A-Z a-z)         
+        echo $v1$v3
+        if test "$temp" = ""; then
+        echo Cannot parse redhat/centos release from /etc/redhat-release 1>&2
+    exit 1
+  fi
+  os_cv_os_ver=$v1$v3
+elif test -f /etc/lsb-release; then
+  . /etc/lsb-release
+  os_cv_os_ver=`echo $DISTRIB_ID|sed 's/^\(.\).*/\1/' | tr A-Z a-z``echo $DISTRIB_RELEASE|sed 's/^\([^.]*\).*/\1/'`
+elif test -f /etc/os-release; then
+  # We'll assume debian
+  . /etc/os-release
+  os_cv_os_ver=d$VERSION_ID
+elif test $HostSystem = darwin -a "`which sw_vers`" != ""; then
+  os_cv_os_ver=`sw_vers -productVersion | sed 's/^\([0-9][0-9]*\.[0-9][0-9]*\).*/\1/' | tr . _`
+elif test "$HostSystem" = linux -a -f /etc/rootfs_version; then
+  os_cv_os_ver=iv
+elif test "$HostSystem" = linux -a "$HostProcessor" = armv7l; then
+  os_cv_os_ver=zynq
+else
+  echo Cannot determine runtime host'!!' 1>&2
+  exit 1
+fi
+		fi
+		)
+    AC_MSG_CHECKING($os_cv_os_ver)
+	dnl check if this is a cross compile session, prepend the sysroot
+	AS_IF([test "x$cross_compiling" = "xyes"],[ 
+	CROSS_SYSROOT=`$CC --print-sysroot`
+	os_os_ver=${CROSS_SYSROOT}${os_cv_os_ver}
+	])
+	])
+AC_SUBST(os_ver, $os_cv_os_ver)
+])
